packages/e2e-tests/test/e2e-tls.spec.ts

```diff
@@ -304,6 +314,73 @@ describe('e2e TLS', function () {
     }
   );
 
+  // Certificate fixtures and general concept mirrors
+  // https://github.com/nodejs/node/blob/1b3420274ea8d8cca339a1f10301d2e80f577c4c/test/parallel/test-tls-client-allow-partial-trust-chain.js
+  // This basically tests that we pass allowPartialTrustChain: true in the TLS options
+  context(
+    'connecting without client cert to server with only partial trust chain',
+    function () {
+      before(function () {
+        // TODO(MONGOSH-1898): Drop Node.js 16 entirely
+        if (process.version.startsWith('v16.')) return this.skip();
+      });
+
+      const server = startTestServer('e2e-tls-partial-trust-chain', {
+        args: [
+          '--tlsMode',
+          'requireTLS',
+          '--tlsCertificateKeyFile',
+          PARTIAL_TRUST_CHAIN_KEY_AND_CERT,
+          '--tlsAllowConnectionsWithoutCertificates',
+          '--tlsCAFile',
+          PARTIAL_TRUST_CHAIN_CA,
+        ],
+      });
+
+      it('works with matching CA (connection string)', async function () {
+        const shell = this.startTestShell({
+          args: [
+            await connectionStringWithLocalhost(server, {
+              tls: 'true',
+              tlsCAFile: PARTIAL_TRUST_CHAIN_KEY_AND_CERT,
+              tlsAllowInvalidHostnames: 'true',
+            }),
+          ],
+        });
+        const result = await shell.waitForPromptOrExit();
+        expect(result.state).to.equal('prompt');
+      });
+
+      it('works with matching CA (system certs)', async function () {
+        if (process.platform !== 'linux') {
+          return this.skip();
+        }
```

Comment:
Are we skipping this test on other platforms because it's more annoying to orchestrate or some other reason?

Reply:
It's because Linux is the only platform where system certificates are loaded from the disk, not through an OS API, so yeah, because it's much harder to "mock" another system CA store on those